src/__tests__/integration-test.ts

```diff
@@ -2,11 +2,81 @@ import * as express from "express";
 import * as request from "supertest";
 import { ApolloServer } from "apollo-server-express";
 import ApolloOpentracing from "../";
-import * as bodyParser from "body-parser";
-import { makeExecutableSchema } from "graphql-tools";
+
+let mockSpanId = 1;
+
+interface MockSpan {
+  id: number;
+  parentId?: number;
+  name: string;
+  options?: any;
+  logs: any[];
+  finished: boolean;
+}
+
+interface MockSpanTree extends MockSpan {
+  children: MockSpanTree[];
+}
+
+const buildSpanTree = (spans: MockSpan[]) => {
```

Comment:
I think an explicit return type might help new people (like me) here to understand the data structure better. But this is more of a nit-pick, no show stopper for me

Reply:
This was done in a hurry. After looking at the code again I think we should actually subclass opentracing.Span with an in memory model. Puts us even closer to the real thing. I actually just wanted to store the children directly on the parent span as this means we don't have to bulid the tree at the end

Reply:
Hmm, nice idea, that might help us make it really neat. Could also be an own module, seems like something worth sharing with the community